src/apps/components/AppAlerts/useAllAppsAlert.ts

```diff
@@ -0,0 +1,73 @@
+import { useUserPermissions } from "@dashboard/auth/hooks/useUserPermissions";
+import {
+  AppFailedPendingWebhooksQuery,
+  EventDeliveryStatusEnum,
+  PermissionEnum,
+  useAppFailedPendingWebhooksQuery,
+} from "@dashboard/graphql";
+import { useMemo } from "react";
+
+type Webhook = NonNullable<
+  NonNullable<AppFailedPendingWebhooksQuery["apps"]>["edges"][0]["node"]["webhooks"]
+>[0];
+interface FailedWebhooksCount {
+  hasFailed: boolean;
+  hasPending: boolean;
+}
+
+const requiredPermissions = [PermissionEnum.MANAGE_APPS];
+
+const defaultFailedWebhooksInfo: FailedWebhooksCount = {
+  hasFailed: false,
+  hasPending: false,
+};
+
+const hasFailedCheck = (webhook: Webhook) =>
+  webhook.failedDelivers && webhook.failedDelivers?.edges?.length > 0;
+const hasPendingCheck = (webhook: Webhook) => {
+  const preliminaryCheck = webhook.pendingDelivers && webhook.pendingDelivers?.edges?.length > 0;
+
+  if (!preliminaryCheck) return false;
+
+  return webhook.pendingDelivers?.edges.some(edge =>
+    edge.node?.attempts?.edges.some(
+      attempt => attempt.node.status === EventDeliveryStatusEnum.FAILED,
+    ),
+  );
+};
+
+export const useAllAppsAlert = (): FailedWebhooksCount => {
```

Comment:
`useAllAppsAlert = (): FailedWebhooksCount` - this signature is a bit odd, the hook refers to some app alerts but the return type mention failed webhook count, so perhaps something like `useWebhookDelivers` is something more meaningful? + rename the type to DeliversInfo?

Reply:
Just `webhooks` would imply webhooks apps you create in Dashboard, here we're checking thirdparty apps